commit 005182cd92ec5dd23fb30e35290cba55bfb0f7ae (refs/changes/46/4846/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-09-18T00:32:27+12:00 (1 year, 1 month ago)
    
    TRITON-773 Error when creating bhyve instance using an image without a requirements attribute

diff --git a/lib/machines.js b/lib/machines.js
index 1b5d721..153afd1 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -423,8 +423,10 @@ function getCreateOptions(req) {
             // With bhyve we also need to always set boot=true for the image
             // disk. This avoids the problem described in OS-6604.
             opts.disks[0].boot = true;
-            if (img.requirements.hasOwnProperty('bootrom')) {
-                opts.bootrom = img.requirements.bootrom;
+
+            var imgBootrom = img.requirements && img.requirements.bootrom;
+            if (imgBootrom) {
+                opts.bootrom = imgBootrom;
             }
         }
 
